plugins/aws_ec2/icon_aws_ec2/connection/connection.py

```diff
@@ -4,17 +4,50 @@
 
 # Custom imports below
 import botocore.session
+import boto3
+import uuid
 from icon_aws_ec2.util.common import ActionHelper
+from logging import Logger
 
 
 class Connection(insightconnect_plugin_runtime.Connection):
     client = None
     helper = ActionHelper()
+    logger: Logger
 
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
 
+    def try_to_assume_role(self, params):
+        if not params.get(Input.ROLE_ARN, ""):
+            return
+
+        session_name = str(uuid.uuid1())
+        assumed_role_object = self.sts_client.assume_role(
```

Comment:
Should we handle any exceptions thrown by the assume_role method?

-  STS.Client.exceptions.MalformedPolicyDocumentException
- STS.Client.exceptions.PackedPolicyTooLargeException
- STS.Client.exceptions.RegionDisabledException
- STS.Client.exceptions.ExpiredTokenException